src/openms/source/ANALYSIS/ID/BasicProteinInferenceAlgorithm.cpp

```diff
@@ -235,8 +219,8 @@ namespace OpenMS
         //Note: the above does not add singleton groups to graph
         ibg.resolveGraphPeptideCentric(true);
         ibg.annotateIndistProteins(true); // this does not really add singletons since they are not in the graph
-        IDFilter::updateProteinGroups(prot_run.getIndistinguishableProteins(), prot_run.getHits());
         IDFilter::removeUnreferencedProteins(cmap, include_unassigned);
+        IDFilter::updateProteinGroups(prot_run.getIndistinguishableProteins(), prot_run.getHits());
```

Comment:
does this change anything?

Reply:
Yes fixes the bug reported by Yasset. Of course you need to remove first, and then update the groups.